src/cmd/package.go

```diff
@@ -336,6 +328,31 @@ func (o *packageMirrorResourcesOptions) run(cmd *cobra.Command, args []string) (
 		err = errors.Join(err, pkgLayout.Cleanup())
 	}()
 
+	// We don't yet know if the targets are internal or external
+	c, _ := cluster.New(ctx) //nolint:errcheck
+
+	if pkgConfig.InitOpts.RegistryInfo.Address == "" {
+		// if empty flag & zarf state available - execute
+		// otherwise return error
+		state, err := c.LoadState(ctx)
+		if err != nil {
+			return fmt.Errorf("no registry URL provided and no zarf state found")
+		}
+		// default registry address will be 127.0.0.1:31999 - we need to update this in order to tunnel
+		state.RegistryInfo.Address = "http://zarf-docker-registry.zarf.svc.cluster.local:5000"
```

Comment:
This will actually still work as with localhost nodeport addresses, so you can keep using what is in state. This way this will work with external registries as well